src/common/security/rbac/context.go

```diff
@@ -59,6 +59,11 @@ func (s *SecurityContext) IsSysAdmin() bool {
 
 // HasReadPerm returns whether the user has read permission to the project
 func (s *SecurityContext) HasReadPerm(projectIDOrName interface{}) bool {
+	// project does not exist
+	if !s.pm.Exist(projectIDOrName) {
+		return false
+	}
+
```

Comment:
what would pm.IsPublic() return if the project does not exist? wouldn't it be false already?